integration-test/test_api.py

```diff
@@ -0,0 +1,125 @@
+import json
+import logging
+import os
+from pathlib import Path
+
+import requests
+from deepdiff import DeepDiff
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(os.environ.get("LOG_LEVEL", logging.INFO))
+
+
+BASE_URL = os.environ.get("BASE_URL", "http://localhost:8008")
+
+
+def actual_status_code_is_expected_status_code(actual_response, expected_status_code):
+    assert actual_response.status_code == expected_status_code
+
+
+def actual_response_is_expected_response(actual_response, expected_path, **kwargs):
+    file_path = Path(Path(__file__).parent, expected_path).resolve()
+    with open(file_path) as file:
+        expected_json = json.load(file)
+
+    diff = DeepDiff(expected_json, actual_response.json(), ignore_order=True, **kwargs)
+    assert diff == {}
+
+
+def test_get_all_collections():
+    actual_response = requests.get(url=BASE_URL + "/collections")
+
+    actual_status_code_is_expected_status_code(actual_response, 200)
+    actual_response_is_expected_response(
+        actual_response, "response/capabilities/200/all_collections.json", exclude_regex_paths=r".*\['href'\]$"
```

Comment:
At the moment href in the JSON is ignored, because when running the tests locally the hostname is localhost while running in the docker it is api. Possible other options than ignoring are:

- Custom comparator for href, which only checks the URL path.
- Change it to api and do not run it locally as docker compose.